services/spar/test-integration/Test/Spar/ScimSpec.hs

```diff
@@ -139,61 +144,113 @@ specUsers = describe "operations with users" $ do
+            -- such a way that it always tries to set the name and handle, it
+            -- might fail because e.g. the handle is "already claimed".
+            --
+            -- Create a user via SCIM
+            user <- randomScimUser
+            (tok, _) <- registerIdPAndScimToken
+            storedUser <- createUser tok user
+            let userid = scimUserId storedUser
+            -- Overwrite the user with another randomly-generated user who has
+            -- the same name and handle
+            user' <- randomScimUser <&> \u ->
+                u { Scim.User.userName = Scim.User.userName user
+                  , Scim.User.displayName = Scim.User.displayName user
+                  }
+            updatedUser <- updateUser tok userid user'
+            -- Get the updated user and check that it matches the user returned
+            -- by 'updateUser'
+            storedUser' <- getUser tok userid
+            liftIO $ updatedUser `shouldBe` storedUser'
+            -- Check that the updated user also matches the data that we sent
+            -- with 'updateUser'
+            liftIO $ do
+                Scim.value (Scim.thing storedUser') `shouldBe` user'
+                Scim.id (Scim.thing storedUser') `shouldBe` Scim.id (Scim.thing storedUser)
+                let meta  = Scim.meta storedUser
+                    meta' = Scim.meta storedUser'
+                Scim.version meta `shouldNotBe` Scim.version meta'
+                Scim.resourceType meta `shouldBe` Scim.resourceType meta'
+                Scim.created meta `shouldBe` Scim.created meta'
+                Scim.location meta `shouldBe` Scim.location meta'
 
         it "updates 'SAML.UserRef' in spar" $ do
-            env <- ask
-            user <- randomSCIMUser
-            (tok, (_, _, idp)) <- registerIdPAndSCIMToken
+            -- Create a user via SCIM
+            user <- randomScimUser
+            (tok, (_, _, idp)) <- registerIdPAndScimToken
             storedUser <- createUser tok user
-            user' <- randomSCIMUser
             let userid = scimUserId storedUser
-            putUser_ (Just tok) (Just userid) user' (env ^. teSpar)
-                !!! const 200 === statusCode
-            vuser' <- either (error . show) pure $ validateSCIMUser' (Just idp) user'
+            -- Overwrite the user with another randomly-generated user
+            user' <- randomScimUser
+            _ <- updateUser tok userid user'
+            vuser' <- either (error . show) pure $ validateScimUser' (Just idp) user'
             muserid' <- runSparCass $ Data.getUser (vuser' ^. vsuSAMLUserRef)
             liftIO $ do
                 muserid' `shouldBe` Just userid
 
-        it "maps ValidSCIMUser to Brig.User completely and correctly (including 'SAML.UserRef')." $ do
+        -- See validateScimUser' for why we don't allow this
+        it "doesn't allow setting emails or phone numbers" $ do
```

Comment:
This test is new. Git mixed it up quite a bit so you might just want to read the two tests without the diff: https://github.com/wireapp/wire-server/blob/fisx-resolve-scim-todos/services/spar/test-integration/Test/Spar/ScimSpec.hs#L225-L253.